### PR TITLE
Trigger Tableau TDVT in connectors on snapshot publish and release (CON-264)

### PR DESCRIPTION
**Pull request by ShimonSte**
Tableau users install the latest ClickHouse JDBC driver manually, so a snapshot or release regression reaches them with no connector release in between. After a publish, fire a `repository_dispatch` to `ClickHouse/connectors` so its Tableau TDVT suite runs against the new driver. Tracking: CON-264.

- **`nightly.yml`:** dispatch `event_type=jdbc-snapshot`; version derived from the pom `<revision>` property (`env.CHC_VERSION` is commented out).
- **`release.yml`:** dispatch `event_type=jdbc-release` with the release `version` input.
- Both are `continue-on-error` so a dispatch failure never fails the publish, and require `secrets.CONNECTORS_DISPATCH_TOKEN` (Contents: read/write on connectors only).

### CODE REVIEW

**Comment by chernser** [comment]
@ShimonSte 

Thank you for the PR!

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -88,3 +88,17 @@ jobs:
             clickhouse-r2dbc/target/clickhouse*.jar
             client-v2/target/client-v2*.jar
             jdbc-v2/target/jdbc-v2*.jar
+      # Tableau users install the latest JDBC driver manually, so a release
```
Comment (chernser) [inline .github/workflows/release.yml]: Release workflow does only push to staging and library may not be accessible for some time after that. 
I think, we need somehow grab latest from maven central and test every week. 

Idea is:
- if something broken - we should catch it on nightly
- if we doing a release and it is a release candidate - then latest should catch but actually we may trigger tests earlier.

Comment (ShimonSte) [reply .github/workflows/release.yml]: Going back to this (the last thing that needs to be done :-) )
You don't think we should check the release before pushing to staging? can we reuse the jar built locally on the workflow machine before being pushed?

Comment (chernser) [reply .github/workflows/release.yml]: @ShimonSte 
We should check release before considering release. But release workflow is a separate task that should have minimal dependencies otherwise we may get blocked. 

Reuse jar where?
We cannot build anything locally and publish. Jar must be built from protected branch and signed by key few people can access. Otherwise it is not a trusted binary.

File: .github/workflows/nightly.yml [hunk 2]
```diff
@@ -63,6 +65,12 @@ jobs:
           #   -e 's|^\(    <version>\).*\(</version>\)$|\1${{ env.CHC_VERSION }}-SNAPSHOT\2|' \
           #   -e 's|${parent.groupId}|com.clickhouse|g' -e 's|${project.parent.groupId}|com.clickhouse|g' '{}' \;
           find . -type f -name "simplelogger.*" -exec rm -fv '{}' \;
+      - name: Record published snapshot version
+        id: publish-version
+        run: |
+          VERSION=$(sed -n 's|.*<revision>\(.*\)</revision>.*|\1|p' pom.xml | head -1)
```
Comment (chernser) [inline .github/workflows/nightly.yml:71]: btw, we have moved to `VERSION` file to simplify such things.

**Review by chernser — APPROVED** [review]

File: .github/workflows/nightly.yml [hunk 3]
```diff
@@ -84,3 +92,29 @@ jobs:
...
+          owner: ${{ github.repository_owner }}
+          repositories: "clickhouse-tableau-tdvt"
+      - name: Run Tableau TDVT and mirror its result
+        env:
+          GH_TOKEN: ${{ steps.tdvt-token.outputs.token }}
+        run: |
+          python3 .github/scripts/mirror_tdvt.py \
+            --event-type jdbc-snapshot \
+            --version "${{ needs.nightly.outputs.jdbc_version }}" \
+            --correlation-id "${{ github.repository }}#${{ github.run_id }}-${{ github.run_attempt }}"
```
[the remark this answers is not in the document]
Comment (chernser) [reply .github/workflows/nightly.yml:120]: This is fine :-) 
We need to know it is failed but we cannot block the publish. Will fix it later.

File: .github/scripts/mirror_tdvt.py [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+    if len(corr) == 1:
+        return corr[0]["id"]
+    if len(corr) > 1:
+        return None
+    if len(matches) > 1:
+        return None
+    if len(matches) == 1:
+        new_runs = [run for run in runs if run.get("id", 0) > threshold]
+        if len(new_runs) > 1:
+            return matches[0]["id"]
```
[the remark this answers is not in the document]
Comment (chernser) [reply .github/scripts/mirror_tdvt.py]: @ShimonSte what do you think?

File: .github/workflows/nightly.yml (hunk 2 above)
Comment (ShimonSte) [reply .github/workflows/nightly.yml:71]: I think the nightly creates its version from the revision tag in the `pom.xml` (which is a version behind the one in the `VERSION` file).
Correct me if i'm wrong.

File: .github/scripts/mirror_tdvt.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (ShimonSte) [reply .github/scripts/mirror_tdvt.py]: This is an edge case where we run multiple tdvt tests at once (shouldn't happen)
Anyway, i changed the tdvt tests so now we get the ID and compare it exactly.

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -85,3 +85,29 @@ jobs:
...
+          private-key: ${{ secrets.WORKFLOW_AUTH_PUBLIC_PRIVATE_KEY }}
+          owner: ${{ github.repository_owner }}
+          repositories: "clickhouse-tableau-tdvt"
+      - name: Run Tableau TDVT and mirror its result
+        env:
+          GH_TOKEN: ${{ steps.tdvt-token.outputs.token }}
+        run: |
+          python3 .github/scripts/mirror_tdvt.py \
+            --event-type jdbc-release \
+            --version "${{ github.event.inputs.version }}" \
```
[the remark this answers is not in the document]
Comment (ShimonSte) [reply .github/workflows/release.yml]: This is correct, fixed it

File: .github/workflows/nightly.yml (hunk 2 above)
Comment (chernser) [reply .github/workflows/nightly.yml:71]: these versions must be synced. but you are right - lets take from true source.
